misk-hibernate/src/main/kotlin/misk/hibernate/SecretColumnType.kt

```diff
@@ -0,0 +1,93 @@
+package misk.hibernate
+
+import com.google.crypto.tink.Aead
+import misk.crypto.AeadKeyManager
+import misk.crypto.KeyNotFoundException
+import misk.logging.getLogger
+import org.hibernate.HibernateException
+import org.hibernate.engine.spi.SharedSessionContractImplementor
+import org.hibernate.usertype.ParameterizedType
+import org.hibernate.usertype.UserType
+import java.io.Serializable
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Types
+import java.util.Properties
+import java.security.GeneralSecurityException
+import org.hibernate.type.spi.TypeConfiguration
+import org.hibernate.type.spi.TypeConfigurationAware
+import java.util.Objects
+
+internal class SecretColumnType : UserType, ParameterizedType, TypeConfigurationAware {
+  companion object {
+    const val FIELD_ENCRYPTION_KEY_NAME: String = "key_name"
+    val logger = getLogger<SecretColumnType>()
+  }
+  private lateinit var aead: Aead
+  private lateinit var keyName: String
+  private lateinit var _typeConfiguration: TypeConfiguration
+
+  override fun setTypeConfiguration(typeConfiguration: TypeConfiguration) {
+    _typeConfiguration = typeConfiguration
+  }
+
+  override fun getTypeConfiguration(): TypeConfiguration = _typeConfiguration
+
+  override fun setParameterValues(parameters: Properties) {
+    val keyManager = _typeConfiguration.metadataBuildingContext.bootstrapContext.serviceRegistry.injector
```

Comment:
Guessing we can't directly pass this in somehow?

Reply:
Yes, hibernate has a different injector being setup, so if you just `@Inject keyManager: AeadKeyManager` you'll get an empty one.